server/routerlicious/packages/services-core/src/checkpointService.ts

```diff
@@ -102,6 +102,10 @@ export class CheckpointService implements ICheckpointService {
 		localCheckpointEnabled: boolean,
 		writeToLocalOnFailure: boolean = false,
 	) {
+		const globalCheckpointErrorMetric = Lumberjack.newLumberMetric(
```

Comment:
Looks like this should only be defined if we enter the `if (writeToLocalOnFailure && localCheckpointEnabled) {` codepath within the globalCheckpointError catch. Otherwise, we'll never complete this for most "happy path" cases

Reply:
I updated this so we only define the metric within that specific case